osquery/remote/http_client.cpp

```diff
@@ -203,8 +203,12 @@ void Client::createConnection() {
 
 void Client::encryptConnection() {
   boost::asio::ssl::context ctx{boost::asio::ssl::context::sslv23};
+  const auto remote_hostname = client_options_.remote_hostname_->c_str();
```

Comment:
Nit: lets make this a normal std::string, the `rfc2818_verification` constructor takes a string; I would then pass `.c_str()` to `SSL_set_tlsext_host_name`.